spyder/plugins/application/plugin.py

```diff
@@ -151,6 +147,11 @@ def on_mainwindow_visible(self):
             container.give_updates_feedback = False
             container.check_updates(startup=True)
 
+        # Hide status bar widget for updates if it doesn't need to be visible.
+        # Note: This can only be done at this point to take effect.
+        if not self.application_update_status.isVisible():
+            self.application_update_status.setVisible(False)
```

Comment:
I don't understand the purpose of this. If the status widget is not visible, then you will set it to not visible? That seems unnecessary. What am I missing?

Reply:
Good question! I'm afraid I don't have a proper answer other than saying that this is what actually makes Qt hide the widget (it took me more than hour to find it).

My guess is that Qt establishes the visibility state of widgets when the app is actually visible on the screen and not before. So, in code `application_update_status` is hidden (which is set in the container for a non-app installation), but we need this extra step to make sure the widget is not visible.

Reply:
Perhaps a clearer option would be to really hide the widget here (and not in the container) instead of having this strange piece of code. What do you think?

Reply:
Well, I don't feel too strongly about it here. I just didn't understand it, so I thought it could be an oversight. However, if it achieves the desired affect, then I call it good. I want to invest as little effort as possible into 5.x.